### PR TITLE
Missing conditional on ServiceMonitor

### LINKED ISSUE #7066: Helm ServiceMonitoring cannot be disabled
**Describe the bug**

In the `3.0.0` helm chart, the monitoring resource `ServiceMonitor` is created by default. Not all Kubernetes clusters use the `monitoring.coreos.com/v1` API and so the resource fails to install. 

The helm `values.yaml` does contain `monitoring.serviceMonitor.enabled` so I assume this conditional was missed off within the ServiceMonitoring resource. 

**To Reproduce**
Install the helm chart with the default values to a cluster without `monitoring.coreos.com/v1` API

**Expected behavior**
You should be able to disable the need to install `ServiceMonitor` when it's not applicable 

PR Incoming shortly

### PR DESCRIPTION
**Pull request by adamstrawson**
**What this PR does / why we need it**:

Affects the Loki Helm chart

The ServiceMonitor resource was missing a conditional for the `enabled` flag, as per the other monitoring resources.

**Which issue(s) this PR fixes**:
Fixes #7066

**Checklist**
- [ ] Documentation added
- [ ] Tests updated
- [ ] Is this an important fix or new feature? Add an entry in the `CHANGELOG.md`.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: production/helm/loki/templates/monitoring/servicemonitor.yaml [hunk 1]
```diff
@@ -1,5 +1,6 @@
----
+{{- if .Values.monitoring.serviceMonitor.enabled }}
```
Comment (trevorwhitney) [inline production/helm/loki/templates/monitoring/servicemonitor.yaml:1]: nit: if we move that down a line, we can just say `{{ if .enabled }}` which might read a tad nicer, wdyt?

**Review by trevorwhitney — APPROVED** [review]
one small nit and please bump the chart version as well, thanks for the fix!

File: production/helm/loki/templates/monitoring/servicemonitor.yaml (hunk 1 above)
Comment (adamstrawson) [reply production/helm/loki/templates/monitoring/servicemonitor.yaml:1]: Happy to make the change, although it wouldn't be consistent with the other monitoring resources? Thoughts?

https://github.com/grafana/loki/blob/75648288ac1dbf2b39a81f47dac0def381399b33/production/helm/loki/templates/monitoring/grafana-agent.yaml#L1
https://github.com/grafana/loki/blob/75648288ac1dbf2b39a81f47dac0def381399b33/production/helm/loki/templates/monitoring/prometheus-rules.yaml#L1

Comment (trevorwhitney) [reply production/helm/loki/templates/monitoring/servicemonitor.yaml:1]: fixed: https://github.com/grafana/loki/pull/7096

Comment (adamstrawson) [reply production/helm/loki/templates/monitoring/servicemonitor.yaml]: Nice one, I was happy to include those changes but wanted the go ahead first, so thanks for that! 😅  

I've updated the conditional to match your PR, once yours is merged I'll bump the chart versions here too.

**Pull request merged**